File: 07_Others/04_BigNum/01_High-precision.cpp
import java.io.*;
public class Main {
    static StreamTokenizer in = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
    static  PrintWriter out = new PrintWriter(new OutputStreamWriter(System.out));
    public static void main(String args[]) throws Exception {
        int a=getInt();
        out.println(a);
        out.flush();
    }
    static int getInt() throws IOException{
        in.nextToken();
        return (int) in.nval;
    }
}

File: 07_Others/04_BigNum/01_High-precision.java
import java.io.*;
import java.math.BigInteger;
  
public class Main {
    public static void main(String args[]) throws IOException {
        StreamTokenizer in = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
        PrintWriter out = new PrintWriter(new OutputStreamWriter(System.out));
 
        BigInteger a;
        BigInteger b;
        in.nextToken();
        int A=(int) in.nval;
        in.nextToken();
        int B=(int) in.nval;
 
        a=BigInteger.valueOf(A);
        b=BigInteger.valueOf(B);
        out.println(a.pow(B).subtract(b.pow(A)));
        out.flush();
    }
}
